core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -93,7 +93,7 @@ private[history] class FsHistoryProvider(conf: SparkConf) extends ApplicationHis
    */
   private def getRunner(operateFun: () => Unit): Runnable = {
     new Runnable() {
-      override def run() = Utils.logUncaughtExceptions {
+      override def run() = Utils.tryOrExit {
```

Comment:
FsHistoryProvider is the module used by history server process to render the application logs....

this getRunner is called for periodically check updated logs and clean logs...

the point here is if anything uncaught is thrown, like OOM, HistoryServer process should be restarted, since it has been unfunctional...